src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/SubsetParameters.java

```diff
@@ -0,0 +1,47 @@
+package au.org.emii.gogoduck.worker;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import au.org.emii.gogoduck.worker.SubsetParameters.SubsetParameter;
+
+public class SubsetParameters extends HashMap<String, SubsetParameter> {
```

Comment:
So when iterating on the map, I use `unmodifiableMap(this)`, which returns a new copy of the map. Speaking of which, also a simple copy of the map or simply making the function `synchronized` will work.

Looking at that again, I'll just make the function synchronized. It's simpler and less likely to cause eyebrows to raise.